precovery/precovery_db.py

```diff
@@ -605,13 +406,16 @@ def create(
 
     def precover(
         self,
-        orbit: AdamOrbits,
+        orbit: Orbits,
         tolerance: float = 30 * ARCSEC,
         start_mjd: Optional[float] = None,
         end_mjd: Optional[float] = None,
         window_size: int = 7,
         datasets: Optional[set[str]] = None,
-    ) -> Tuple[List[PrecoveryCandidate], List[FrameCandidate]]:
+        propagator: Optional[
+            Propagator
+        ] = None,  # should we initialize an assist propagator?
```

Comment:
```suggestion
    propagator: Optional[Type[Propagator]] = None
```

Accept a propagator class. Then inside the function we initialize it.